athena-teradata/src/main/java/com/amazonaws/athena/connectors/teradata/TeradataMetadataHandler.java

```diff
@@ -244,7 +244,7 @@ private boolean useNonPartitionApproach(GetTableLayoutRequest getTableLayoutRequ
     {
         final String getPartitionsCountQuery = "Select  count(distinct partition ) as partition_count FROM " + getTableLayoutRequest.getTableName().getSchemaName() + "." +
                 getTableLayoutRequest.getTableName().getTableName() + " where 1= ?";
-        String partitioncount = configOptions.get("partitioncount");
+        String partitioncount = configOptions.containsKey("partition_count") ? configOptions.get("partition_count") : configOptions.getOrDefault("partitioncount", "500");
```

Comment:
where is the 500 partition from? and should this be a static of configurable value?